Tavisca.Training2017.HandsOn.Assignment3/Tavisca.Training2017.HandsOn.Assignment3/Solution1.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Tavisca.Training2017.HandsOn.Assignment3
+{
+
+    /*
+                                                       Problem Statement 1.
+   Write a program to display user’s complete mailing address. Accept user’s name, city, street, pin and house no. and store it in a variable and display it.
+   */
+
+    class Solution1
+    {
+        Address mailingAddress;
+        public Solution1()
+        {
+            mailingAddress = new Address();
+        }
+
+        public void InputInformation()
+        {            
+            mailingAddress.Name = InputType<String>("Enter User Name:");
+            mailingAddress.City = InputType<String>("Enter City:");
+            mailingAddress.Street = InputType<String>("Enter Street:");
+            mailingAddress.Pin = InputType<String>("Pin:");
+            mailingAddress.HouseNo = InputType<String>("House No.:");
+        }
+
+        public void DisplayMailingAddress()
+        {
+            InputType<String>(mailingAddress.ToString());
+        }
+
+        private string InputType<T>(T type)
```

Comment:
Unnecessary / incorrect use of generics. Generics are used when you would want to handle different types with same code, not when you want to handle different strings.